pkg/state/cost/cost.go

```diff
@@ -213,42 +212,40 @@ func (cc *ClusterCost) UpdateNodeClaim(ctx context.Context, nodeClaim *v1.NodeCl
 	if nodeClaimMissingLabels(*nodeClaim) {
 		// not technically a failure mode as we expect to retry once the
 		// labels are propagated
-		return
+		return nil
 	}
 
 	np := &v1.NodePool{}
 	if err := cc.client.Get(ctx, types.NamespacedName{Name: nodeClaim.Labels[v1.NodePoolLabelKey]}, np); err != nil {
-		log.FromContext(ctx).Error(serrors.Wrap(err, "nodepool", nodeClaim.Labels[v1.NodePoolLabelKey], "nodeclaim", klog.KObj(nodeClaim)), "failed to process nodeclaim for cost tracking")
 		failed = true
-		return
+		return serrors.Wrap(err, "nodepool", nodeClaim.Labels[v1.NodePoolLabelKey], "nodeclaim", klog.KObj(nodeClaim))
```

Comment:
when did we start using klog to get namespaced names? I am more familiar with client.ObjectKeyFromObject or https://github.com/awslabs/operatorpkg/blob/main/object/object.go#L34.